src/mongo/util/assert_util.h

```diff
@@ -353,3 +353,17 @@ std::string demangleName(const std::type_info& typeinfo);
                                           ::mongo::logger::LogSeverity::Log())                 \
             << "caught unknown exception in destructor (" << __FUNCTION__ << ")" << std::endl; \
     }
+
+/**
+ * The purpose of this macro is to instruct the compiler that a line of code will never be reached.
+ *
+ * Example:
+ *     // code above checks that expr can only be FOO or BAR
+ *     switch (expr) {
+ *     case FOO: { ... }
+ *     case BAR: { ... }
+ *     default:
+ *         MONGO_UNREACHABLE;
+ */
+
+#define MONGO_UNREACHABLE ::mongo::invariantFailed("Hit a MONGO_UNREACHABLE!", __FILE__, __LINE__);
```

Comment:
I thought about it, and I think yes, because now that this is in the v3.0 branch, future backports from the server may include code that uses it.